### ISSUE TITLE
Auth-context available for use in export-config

### ISSUE THREAD

**Issue by DennisGlindhart** [open-telemetry/opentelemetry-collector#4814]
When using a (Server)Authentication extension for OTEL-Collector (i.e. basic auth, oidc etc. from contrib), I would like it to be possible to somehow extract the username (or other auth-data/claim) and use it in i.e. exporter-headers.

The auth context seems to be returned from the extension in authenticate-method ( https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/oidcauthextension/extension.go#L136 ) so it should already be available in the context.

**Describe the solution you'd like**
Ability to use the auth context in exporter-config.
```
exporters:
  otlp:
    headers:
      x-scope-orgid: <USERNAME FROM AUTH>
```

**Additional context**
One use case here could be to have only a singe OTEL-collector running on multitenant server/cluster-node etc and let software authenticate with the OTEL-collector and be exported with Multitentenancy-header (in this example Scope-header used Grafana Tempo/Loki).

**Comment by jpkrohling** [comment]
@bogdandrutu, @tigrannajaryan, do you think we could change the confighttp to accept a notation like `context.user_id`, which would make it look up the `user_id` key within the context? Or perhaps you have another suggestion? In any case, I would do it after GA.

**Comment by tigrannajaryan** [comment]
>use it in i.e. exporter-headers.

@jpkrohling how will this work? We don't maintain the context in batch processor or in queued_retry, so the incoming context won't be available in the exporters, right?

**Comment by jpkrohling** [comment]
We don't yet, but I do plan on working on that soon (#4544).

**Comment by bogdandrutu** [comment]
@tigrannajaryan we do maintain the context in `queued_retry` only in batching we have a problem, since we batch requests from multiple sources.

**Comment by kovrus** [comment]
@DennisGlindhart does #12892 resolve this issue for you?

**Comment by DennisGlindhart** [comment]
@kovrus Reading the README it would seem so, yes.

This configuration should fill the username from auth into User-ID header when sending to exporter if I understand correctly.
```
extensions:
  headers_setter:
    headers:
      - key: User-ID
        from_context: "user_id"
```

There is still the batch-limitation (Nice to have), but I guess in the future relying on the same work mentioned earlier could solve that? (https://github.com/open-telemetry/opentelemetry-collector/issues/4544).

**Comment by kovrus** [comment]
> There is still the batch-limitation (Nice to have), but I guess in the future relying on the same work mentioned earlier could solve that? (https://github.com/open-telemetry/opentelemetry-collector/issues/4544).
That's correct. See https://github.com/open-telemetry/opentelemetry-collector/issues/5733#issuecomment-1204648704, it could be potentially temporal workaround.
